lib/ansible/modules/cloud/opennebula/one_vm.py

```diff
@@ -1220,8 +1223,8 @@ def get_connection_info(module):
     if not password:
         password = os.environ.get('ONE_PASSWORD')
 
-    if not(url and username and password):
-        module.fail_json(msg="One or more connection parameters (api_url, api_username, api_password) were not specified")
+    if not url:
+        module.fail_json(msg="Opennebula API url (api_url) is not specified")
     from collections import namedtuple
 
     auth_params = namedtuple('auth', ('url', 'username', 'password'))
```

Comment:
Just extract `ONE_AUTH` env var and add it here in this tuple

Reply:
As mentioned previous comment, python-oca will do that.